### PR TITLE
app/admin-conky-1.10.8-r9: Patched crash on virtual Pulseaudio sinks

### PR DESCRIPTION
**Pull request by jeeger**
Using upstream commits
https://github.com/brndnmtthws/conky/commit/2a7226d89331f7d5d0abf0b9aff6d75435dd3a2d and
https://github.com/brndnmtthws/conky/commit/ddf8a0f58f3110e31ec4c44ecff838d2b231f6f3. Added missing
braces for multi-line macro.

Signed-off-by: Jan Seeger <jan.seeger@thenybble.de>

### CODE REVIEW

File: app-admin/conky/files/conky-1.10.8-virtual-sinks.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+@@ -53,8 +53,10 @@ void pa_sink_info_callback(pa_context *c, const pa_sink_info *i, int eol, void *
+         pdr->sink_mute = i->mute;
+         pdr->sink_card = i->card;
+         pdr->sink_index = i->index;
+-        pdr->sink_active_port_name.assign(i->active_port->name);
+-        pdr->sink_active_port_description.assign(i->active_port->description);
++		if (i->active_port != nullptr) {
++			pdr->sink_active_port_name.assign(i->active_port->name);
++			pdr->sink_active_port_description.assign(i->active_port->description);
++		}
```
Comment (billie80) [inline app-admin/conky/files/conky-1.10.8-virtual-sinks.patch:14]: No expert on this but what about the else clause of the original commit brndnmtthws/conky@2a7226d where sink_active_port_name and sink_active_port_description are erased?

**Comment by jeeger** [comment]
What do you mean by "lastest changes"? The fix is already in upstream code, and the 1.11 revision of conky is blocked since Lua has been stuck on 5.1 for ages.

File: app-admin/conky/files/conky-1.10.8-virtual-sinks.patch (hunk 1 above)
Comment (jeeger) [reply app-admin/conky/files/conky-1.10.8-virtual-sinks.patch:14]: Good point, this might leave outdated information in the buffer. I'll add it too.

**Comment by billie80** [comment]
Sorry I meant the latest change in the ebuild. I switched from cmake-utils to cmake eclass yesterday!

**Comment by jeeger** [comment]
Ah, okay, I'll rebase!

**Comment by billie80** [comment]
Thanks! Can you please squash this into one commit and make a new revision with app/admin-conky-1.10.8-r10 so everyone is forced to update and benefits from the changes?

**Comment by jeeger** [comment]
Oh, right. Sorry, forgot about the bump!

**Comment by billie80** [comment]
One last thing. You need to remove the patch from the r9 version!

**Comment by jeeger** [comment]
Done!

**Comment by billie80** [comment]
Pushed as 8ed09d7! Thanks for the patches!

**Pull request closed without merging**